go/vt/schemadiff/env.go

```diff
@@ -0,0 +1,28 @@
+package schemadiff
+
+import (
+	"vitess.io/vitess/go/mysql/collations"
+	"vitess.io/vitess/go/vt/sqlparser"
+)
+
+type Environment struct {
+	CollationEnv *collations.Environment
+	DefaultColl  collations.ID
+	Parser       *sqlparser.Parser
+}
+
+func NewTestEnv() *Environment {
+	return &Environment{
+		CollationEnv: collations.MySQL8(),
+		DefaultColl:  collations.MySQL8().DefaultConnectionCharset(),
+		Parser:       sqlparser.NewTestParser(),
+	}
+}
+
+func NewEnv(collEnv *collations.Environment, defaultColl collations.ID, parser *sqlparser.Parser) *Environment {
+	return &Environment{
+		CollationEnv: collEnv,
+		DefaultColl:  defaultColl,
+		Parser:       parser,
+	}
+}
```

Comment:
I predict it might be useful: let's add a `NewMySQL8Env()` function that uses `utf8mb4` and possibly also creates a MySQL8 compliant parser.

Reply:
@shlomi-noach I'm hesitant actually about that and I like that this is explicit and the caller is required to pass these in. We also explicitly name the default one as a test specific one.

The main reason for this is also slightly less collations, but also the parser since the parser has MySQL specific version behavior, so you almost always want to pass in a specific version parser from the context you're in.

So I very much prefer not to add this and see that if we seem to have a need, to look at that and do the proper thing at that point.